tsfc/driver.py

```diff
@@ -100,7 +100,10 @@ def compile_integral(idata, fd, prefix, parameters):
 
     mesh = idata.domain
     coordinates = fem.coordinate_coefficient(mesh)
-    funarg, prepare_, expression = prepare_coefficient(integral_type, coordinates, "coords")
+    if mesh.ufl_cell().cellname() in ["interval", "triangle", "tetrahedron"]:
+        funarg, prepare_, expression = prepare_coefficient(integral_type, coordinates, "coords", mode='list_tensor')
```

Comment:
If I understand the description correctly, you want to do this for affine coordinates?  But this is not the correct check for that.